test/SILGen/reabstract-tuple.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-swift-frontend  -emit-silgen -verify %s
+
+// SR-3090:
+
+class Box<T> {
```

Comment:
Shouldn't the test have CHECK: lines and go into one of the existing files where we test reabstraction?

But you can fix that on master, this looks good for 3.0.